scripts/script-free.sh

```diff
@@ -0,0 +1,115 @@
+# Get the machine architecture type
+archtype=$(uname -m)
+
+if [ "$archtype" = "x86_64" ]; then
+    archtype="amd64"
+fi
+
+# Account ID from script argument
+ACCOUNT_ID=$1
+# Account ID from script argument
+DELEGATE_TOKEN=$2
+DELEGATE_IMAGE_VERSION=$3
+# Check the operating system type and run commands accordingly
+if [ "$ostype" = "Darwin" ]; then
+    # Commands for macOS
+    echo "Running commands for macOS"
+
+    # Run the Docker container with environment variables
+    docker run -d --cpus=1 --memory=2g --restart=always\
+      -e DELEGATE_NAME=docker-delegate \
+      -e NEXT_GEN="true" \
+      -e DELEGATE_TYPE="DOCKER" \
+      -e ACCOUNT_ID=$ACCOUNT_ID \
+      -e DELEGATE_TOKEN=$DELEGATE_TOKEN \
+      -e LOG_STREAMING_SERVICE_URL=https://app.harness.io/gratis/log-service/ \
+      -e DELEGATE_TAGS="macos-$archtype" \
+      -e RUNNER_URL=http://host.docker.internal:3000 \
+      -e MANAGER_HOST_AND_PORT=https://app.harness.io/gratis harness/delegate:$DELEGATE_IMAGE_VERSION
+
+    # Wait for a while before continuing the script execution
+    echo "Sleeping for 2 seconds"
+    sleep 2
+
+    echo "Removing already existing binary"
+    rm -f harness-docker-runner-linux-$archtype
+
+    # Download the harness-docker-runner for macOS
+    echo "Downloading the runner binary"
+    wget -q https://github.com/harness/harness-docker-runner/releases/latest/download/harness-docker-runner-darwin-$archtype 
+    
+    # Make the downloaded file executable
+    echo "Chmod runner binary"
+    sudo chmod +x harness-docker-runner-darwin-$archtype 
+    
+    # Start the harness-docker-runner server
+    echo "running the binary"
+    sudo ./harness-docker-runner-darwin-$archtype server &
+    
+elif [ "$ostype" = "Linux" ]; then
+    echo "Running commands for Linux"
+
+    # Run the Docker container with environment variables
+    docker run -d --cpus=1 --memory=2g --net=host --restart=always\
+      -e DELEGATE_NAME=docker-delegate \
+      -e NEXT_GEN="true" \
+      -e DELEGATE_TYPE="DOCKER" \
+      -e ACCOUNT_ID=$ACCOUNT_ID \
+      -e DELEGATE_TOKEN=$DELEGATE_TOKEN \
+      -e LOG_STREAMING_SERVICE_URL=https://app.harness.io/gratis/log-service/ \
+      -e DELEGATE_TAGS="linux-$archtype" \
+      -e MANAGER_HOST_AND_PORT=https://app.harness.io/gratis harness/delegate:$DELEGATE_IMAGE_VERSION
+
+    # Wait for a while before continuing the script execution
+    echo "Sleeping for 2 seconds"
+    sleep 2
+
+    echo "Removing already existing binary"
+    rm -f harness-docker-runner-linux-$archtype
+
+    # Download the harness-docker-runner for Linux
+    wget -q https://github.com/harness/harness-docker-runner/releases/latest/download/harness-docker-runner-linux-$archtype 
+    
+    # Make the downloaded file executable
+    sudo chmod +x harness-docker-runner-linux-$archtype 
+    
+    # Start the harness-docker-runner server
+    sudo ./harness-docker-runner-linux-$archtype server &
+else
+    echo "Assuming it's a Windows machine"
```

Comment:
@rajatharanganath Added steps for windows. execution will look like this.
Didn't put ostype = windows condition, because it might and might not return "windows", hence assuming it's a windows machine when it comes to else condition.
<img width="1312" alt="Screenshot 2023-12-19 at 11 17 53 AM" src="https://github.com/harness/harness-docker-runner/assets/99166403/0bf8329d-46b3-4c8f-9ded-df4ea2a419b2">

Reply:
Looks good to go